diesel/src/pg/connection/mod.rs

```diff
@@ -193,15 +193,11 @@ mod tests {
         let connection = connection();
         let query =
             crate::sql_query("SELECT not_existent FROM also_not_there;").execute(&connection);
+
         if let Err(err) = query {
-            match err {
-                DatabaseError(_, string) => {
-                    assert_eq!(Some(26), string.statement_position());
-                }
-                _ => unreachable!(),
+            if let DatabaseError(_, string) = err {
```

Comment:
This test should fail if something other than that error is returned. The old variant ensured this, while this one will silently accept other error messages.